### PR TITLE
[@dvgis/dc-sdk] add type definitions for dc-sdk

### PR DESCRIPTION
**Pull request by cavencj**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/dc-sdk/dc-sdk-tests.ts
1. types/dc-sdk/Cesium.d.ts
1. types/dc-sdk/index.d.ts
Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6ebe71d46a0117aaff4b39f6aaaab99fbe1464a6
</p>

**Review by jakebailey — CHANGES_REQUESTED** [review]
1. Presumably these are actually the types for `@dvgis/dc-sdk`; you have added these to the wrong directory for a scoped package.
2. This is a monumentally large change, but it's also a package you own. Can you just ship these types with your package instead of submitting them here?

**Comment by cavencj** [comment]
@jakebailey 

Thank you for your reply.

I've tried the solution you mentioned, but it's not possible to do code hints in IDEs like vscode and webstorm, IDEs can only use code hints if you put the type definitions under @types, so I'd appreciate it if you could move the type definitions of @dvgis/dc-sdk under @types!

Thank you for your help.

**Comment by jakebailey** [comment]
There are a massive number of packages which are written in TS and publish their own types, or are written in JS and publish their own types. Being on DT doesn't offer a major advantage if your hard are already installing your package. You definitely do not need to be under the `@types` scope to have types.

**Comment by jakebailey** [comment]
(and no matter what, these types do not describe `dc-sdk`, they describe `@dvgis/dc-sdk` and need to be placed in `types/dvgis__cd-sdk`; see the readme for info on scoped packages.)

**Comment by jakebailey** [comment]
The directory is still wrong, it needs to be `dvgis__dc-sdk`.

But, I'm still not convinced that this is a good thing to put on DT. It's 45 thousand lines of declarations for a package you control; nobody except you is going to be able to understand changes to it, and there should be no meaningful reason for it to be on DT when you have the power to ship your types in your own package.

File: types/dvgis_dc-sdk/tslint.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
+        "npm-naming": false,
```
Comment (jakebailey) [inline types/dvgis_dc-sdk/tslint.json:4]: Your package is on npm; the lint is trying to tell you that your types package is named incorrectly.

File: types/dvgis__dc-sdk/dvgis__dc-sdk-tests.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import * as DC from 'dvgis_dc-sdk'
+import * as DC from 'dvgis__dc-sdk'
```
Comment (jakebailey) [inline types/dvgis__dc-sdk/dvgis__dc-sdk-tests.ts:1]: You need to import this as `@dvgis/dc-sdk`.

File: types/dvgis__dc-sdk/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "rules": {
+        "@definitelytyped/dt-header": "off",
+        "@definitelytyped/no-any-union": "off",
+        "@definitelytyped/strict-export-declare-modifiers": "off"
```
Comment (jakebailey) [inline types/dvgis__dc-sdk/.eslintrc.json:5]: These are all lints which should be addressed or discussed, not disabled.

File: types/dvgis__dc-sdk/tslint.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        "prefer-template": false,
+        "unified-signatures": false,
+        "no-internal-module": false,
+        "no-unnecessary-qualifier": false,
+        "no-unnecessary-class":false,
+        "interface-over-type-literal":false,
+        "no-empty-interface":false,
+        "adjacent-overload-signatures":false
+
+    }
```
Comment (jakebailey) [inline types/dvgis__dc-sdk/tslint.json:19]: Please, do not disable these.

**Review by jakebailey — CHANGES_REQUESTED** [review]
You can't disable the lints to fix them; if you don't know how to fix them, then please ask!

But still, I don't think we should merge this in. It's 45 thousand lines that you could just publish in your package.

**Comment by cavencj** [comment]
@jakebailey  

Thank you for your reply.

I went with a different approach to adding the type definition

This PR can be rejected

Thank you.

**Pull request closed without merging**